### PR TITLE
Feat/integers

### PR DESCRIPTION
**Pull request by IceTDrinker**
wopbs API is improvable for sure, as a first step this already allows to review the integration of the integer module

note that I disabled `test_wopbs_bivariate_crt_wopbs_param_message_[34]_carry_[34]` which generate vecs of 550 GB and 2200 TB respectively, I don't know those primitives enough to dig deeper at this time

### CODE REVIEW

**Comment by IceTDrinker** [comment]
Must have mixed something, shortint tests should not be running integet tests

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

**Comment by IceTDrinker** [comment]
will address the &mut comments for integers, we have the same kind of design pattern in shortint, I think we'll do a refactoring pass on this in a further PR

File: tfhe/src/integer/ciphertext/mod.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+//! This module implements the ciphertext structures.
+use crate::shortint::Ciphertext as ShortintCiphertext;
+use serde::{Deserialize, Serialize};
+
+/// Id to recognize the key used to encrypt a block.
+#[derive(Debug, PartialEq, Eq, Copy, Clone, Serialize, Deserialize)]
+pub struct KeyId(pub usize);
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/integer/ciphertext/mod.rs:7]: fixed

[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/integer/ciphertext/mod.rs:7]: It's odd that it does not warn about the type not being used 🤔

[the remark this answers is not in the document]
Comment (tmontaigu) [reply tfhe/src/integer/ciphertext/mod.rs:7]: Since its pub, its part of the public API

[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/integer/ciphertext/mod.rs:7]: yep makes sense

File: tfhe/src/integer/client_key/mod.rs [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+        self.key.parameters
+    }
+
+    /// Encrypts an integer in radix decomposition
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use tfhe::integer::ClientKey;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2;
```
Comment (IceTDrinker) [inline tfhe/src/integer/client_key/mod.rs:79]: very weird pattern

Comment (IceTDrinker) [reply tfhe/src/integer/client_key/mod.rs:79]: meaning: mixed APIs/structs from the user perspective

**Comment by IceTDrinker** [comment]
Normally comments were adressed, I'm looking to make use of the contiguous primitives from core crypto (and I remember why I did not do it at first, it requires a bit of rework and is basically an implementation detail)

**Review by sarah-ek — APPROVED** [review]

**Comment by IceTDrinker** [comment]
hmm looks like a doc test that passes in the CI fails on my computer consistently 🤔

**Comment by IceTDrinker** [comment]
bad keycache, I'll end up adding checksums for the keycache at least

**Comment by IceTDrinker** [comment]
I looked at rewriting the lut generation/format, it should not be too hard as the indices are already managed by the generation function, introducing a "WopbsAccumulator" similar to the Shortint Accumulator should be enough to make it usable for end users, behind the scenes we'll translate the indices to some linear index to store data in the right part of a contiguous container

**Pull request merged**